chain-impl-mockchain/src/testing/builders/block_builder.rs

```diff
@@ -32,41 +32,39 @@ impl GenesisPraosBlockBuilder {
         }
     }
 
-    pub fn with_parent(&mut self, parent: &Header) -> &mut Self {
-        self.with_parent_id(parent.hash());
-        self.with_date(parent.block_date());
-        self.with_chain_length(parent.chain_length());
-        self
+    pub fn with_parent(self, parent: &Header) -> Self {
```

Comment:
Can you describe the benefits of changing this? Both styles have their own advantages.